Code/Framework/AzCore/AzCore/PlatformDef.h

```diff
@@ -281,35 +281,45 @@
+// The non-short link is being used here to not rely on a URL shortener service
+// https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxBIAHKQADqgKhE4MHt6%2BekkpjgJBIeEsUTFc8XaYDmlCBEzEBBk%2Bfly2mPZ5DDV1BAVhkdFxtrX1jVktCsM9wX3FA2UAlLaoXsTI7BzmAMzByN5YANQmm24TxMHAAPoAbniYAO5H2CYaAILPLwD0H/sAatF4NEwCn2ES8dEcDH2wQIZwYKWQwIICCYBH2dUw%2By8CiYEXoUMhTH2aBYCToGMcbCJskwqgI7zQcIINISxHxqKZEwuoPBwQUEHmhwA7FZXvsxVTGczWRN0CAQKdztdbnd9gplqsMUcACKHMxmAASbVoqH2AHUSLR0OYzEcRS9xRKJlLVQRZSA7gqjEr7vs7ngsEJ1WtDpsdQFrYbaMazRarXrbe93g7kMjWVgJiYAKwWTY2zNakzCwsFzZ2h13FN1C6ov1YLVAunZ3NZgtFwUlu1J8VffZsFjIBIATxBYNoEIUibM2yo%2BwuF2RCi5o4hEDn3LHwQufYHg/mXbFa%2BXm%2B3Q4g6YIpFVQcwADp0CimPzL2qVmsbykAF6Yfl70uTzauAC/4ArO85MIu64roePIMBcdwnru%2B6gZBm7wZg/anrWmD1hMl5YYGr63vetRPr6/qYARGrvngX4/gmrxbIBVCJgxordt8fYsKgVwYihcLATOc4LkuMGriJG6wZx3GYHubEHuJEJbuhXE8WeDbPted4PqRL5UZ%2B37zL%2BnZTkxAmgcJfFiXxcFSTxsn2uK0ESTZynSRAWE4ReZEBppxGPos3kUZp%2Bl0X%2BDEmQw%2BDMaxDlij2pz0JCfETuF07meBCnBFZR6wQljD2Q6TmKXlDAQLpb5%2BaFxkAZFQGpSBQkZZZRWoQiiUFY5mWwRWCiJe55GURV2mGfRbwRVFLGxfsPbbggrLJWZjUQTl2UwUp/ZzR18nWbNxBlb5w2XhoGmEdRtEjWFY01RN9WCWBy2iS13W7VtyE5S5G17fhB0kQFx2BYNt4hRd1WmTFDozehyAkiOMEpVdDX3V1q3OQhr1Pet0MJPtp2VQFEZGia5rEJa1onXpNEGUZ/5gwjd0WStGNof2JLo11H1Y/1Pm44d%2BzhnqkbRsTpN6nhA3BZTVU07V0VvHJ%2BzEJgBArJCGijcWk3Qr2TBZfZhZluKDJOqoLJsgrQJeGOIY6hyBBdXy1Pg%2BKivK8QkKKwolt0pdxYcIstCcJmvB%2BBwWikKgnBuNY1hXoRuqbDwpAEJofuLAA1iAmybDeWe53n%2BcAGz6JwkjByn4ecLwCggMdyeh37pBwLAMCICgqAkmSZAUBAxKkvQMTAFwCekFgNxrD8yoAPIJIwnCJzQY7RNXEAROXETBHUg5z7w6/MMQg6TxE2iVHXifEmwgiTwwtBb/XI%2BYKCwBuGItDV9wvBYCwhjAOId/4IrVQeJvzDjSSoXgmTb3IIINo5daB4AiMQTeHgsDlxhHgFg29FhUAMMABQE97jT1nu/GQggRBiHYFIEh8glBqHLroFoBgjAoGjpYfQ8Dq6QEWKgBIHQ34AFoZTalMJYawXBBS8GksQM46Z4CLAqFUZwEBXCjGaKQQI0wiglGyMkVIAgVHaNyGkXomi5itHaNUSY%2BjxhtBPhY7oxj%2BilCGN0Kxzj6gONmKUOR14JD%2B0DmXO%2BEcOD7FULEAufCC6SH2MAZAyB9hDxvFwfYEBcCEBIPHLg8xeB1y0IZUgGdMzHQDhwUupAMGFNICHMOQSq41yTinRYTdW7LAIAkcB5BKC907qEVg6xQnhMidE2J8Ts6J0wPgIg0i9D8FIaIcQlCZnUJUOoO%2B9DSB3EQQkTBxcOBB0qeXIJk9wFtNRKgGc/SIlRJiXEhJSSIAeA7v3DJWT6n1zyQgTATAsAxCfMU0p5TjpVIkZXWwdScmp3yZnbO%2BcYW5yLsUzYATqkgvBY0luTSIBIC6f3DpPd259wGLsb%2BXAzAaGOqPPA48p4zxDvPcES9KCrzvrvTekCWX70PsfBwkDz6MAIFfG%2B5csCP2flGN%2BidP7f1/mHf%2BtigHl1AcgcB6xE7QhgXfOBCCkEYHWGHNBGD35YJwXg6lRD56yDIfM6QizFDLLoSAMw%2Bhv7MJEawzVHCnzcN4ZwARrohEsIsGIiRPEpHkQ9XImxCi/BKMiq49RhRHEGN0ekTwTQk0dA8Vo6x5iBBdBGKmsYZjbG5smJm0xEwXEFtURW9xGjE2ZKWIRXxOy9lAorsEi5gyiVGHiWYG8Gh%2B3JNSZM552SGnpyhTnWFMKdn/JABUttNTQW1waY3dFSAWknNxdi6IPS2CcE7VE7twBe39v7bwcZaSpktEWZaih1rZC2toash16zNnbOKa2g5nAjmtPAfsM5ISwmXKJIwk9JKz0aGSQ8glrIthmBeaixYHyvkDF%2BSXXgAL9mBJBdXFdbyJ1ZyndOvOOzEXYeRRwMdBGdlmCRcCqjrzcmLBDSkZwkggA%3D%3D%3D
+
 #if defined(__has_builtin)
-    #if __has_builtin(__builtin_memcpy) && (!defined(AZ_COMPILER_GCC) || AZ_COMPILER_GCC >= 130000)
+    #if __has_builtin(__builtin_memcpy) && (!defined(AZ_COMPILER_GCC))
```

Comment:
This is the opposite.
The GCC build was failing because it was trying to use `__builtin_memcpy` in a constant evaluation context and it was failing.

So the change is to have GCC13+ use the behavior of GCC12 and below.